### PR TITLE
[scudo] Add -Wconversion for tests and clean-up warnings.

### PR DESCRIPTION
**Pull request by cferris1000**
Fix all the places where the tests are doing implicit conversions.

### CODE REVIEW

File: compiler-rt/lib/scudo/standalone/tests/combined_test.cpp [hunk 1]
```diff
@@ -54,7 +54,7 @@ void checkMemoryTaggingMaybe(AllocatorT *Allocator, void *P, scudo::uptr Size,
     EXPECT_DEATH(
         {
           disableDebuggerdMaybe();
-          reinterpret_cast<char *>(P)[-1] = 0xaa;
+          reinterpret_cast<char *>(P)[-1] = '\xaa';
```
Comment (hctim) [inline compiler-rt/lib/scudo/standalone/tests/combined_test.cpp:57]: Let's just use 'A' (this is just a crash test, so the value doens't matter).

File: compiler-rt/lib/scudo/standalone/tests/combined_test.cpp [hunk 2]
```diff
@@ -63,7 +63,7 @@ void checkMemoryTaggingMaybe(AllocatorT *Allocator, void *P, scudo::uptr Size,
     EXPECT_DEATH(
         {
           disableDebuggerdMaybe();
-          reinterpret_cast<char *>(P)[Size] = 0xaa;
+          reinterpret_cast<char *>(P)[Size] = '\xaa';
```
Comment (hctim) [inline compiler-rt/lib/scudo/standalone/tests/combined_test.cpp:66]: ditto

File: compiler-rt/lib/scudo/standalone/tests/combined_test.cpp [hunk 3]
```diff
@@ -345,7 +345,7 @@ SCUDO_TYPED_TEST(ScudoCombinedTest, ReallocateLargeIncreasing) {
   // we preserve the data in the process.
   scudo::uptr Size = 16;
   void *P = Allocator->allocate(Size, Origin);
-  const char Marker = 0xab;
+  const char Marker = '\xab';
```
Comment (hctim) [inline compiler-rt/lib/scudo/standalone/tests/combined_test.cpp:348]: Maybe just use 'A' here as well, rather than the hex escape prefix?

**Review by hctim — COMMENTED** [review]
Same comment in a few other places, would you be able to change the char-based magic values to something without the hex prefix (like `'A'`)?

File: compiler-rt/lib/scudo/standalone/tests/combined_test.cpp (hunk 1 above)
Comment (cferris1000) [reply compiler-rt/lib/scudo/standalone/tests/combined_test.cpp]: Done.

File: compiler-rt/lib/scudo/standalone/tests/combined_test.cpp (hunk 2 above)
Comment (cferris1000) [reply compiler-rt/lib/scudo/standalone/tests/combined_test.cpp]: Done.

File: compiler-rt/lib/scudo/standalone/tests/combined_test.cpp (hunk 3 above)
Comment (cferris1000) [reply compiler-rt/lib/scudo/standalone/tests/combined_test.cpp]: Done.

**Comment by cferris1000** [comment]
Any further comments?

**Review by hctim — APPROVED** [review]

**Review by ChiaHungDuan — APPROVED** [review]

**Pull request merged**